hub/core/meta/tensor_meta.py

```diff
@@ -76,30 +81,30 @@ def adapt(self, buffer: memoryview, shape: Tuple[int], dtype) -> memoryview:
             TensorDtypeMismatchError: Dtype for array must be equal to or castable to this meta's dtype
             TensorInvalidSampleShapeError: If a sample already exists, `len(array.shape)` has to be consistent for all arrays.
         """
-        dtype = np.dtype(dtype)
+        if isinstance(buffer, np.ndarray):
+            shape = buffer.shape
+            dtype = buffer.dtype
+            array = buffer
```

Comment:
this is probably not a good idea, in my branch i am getting rid of tensor_meta.adapt

Reply:
i am refactoring append/extend input preprocessing (like casting/compressing) which also means updating meta needs to be changed